blockchain/validation.go

```diff
@@ -69,8 +121,51 @@ func (bc *BlockChain) ValidTransaction(t *Transaction) (bool, TransactionCode) {
 	return true, ValidTransaction
 }
 
+// ValidCloudBase returns true if a transaction is a valid CloudBase transaction
+// and false otherwise
+func ValidCloudBase(t *Transaction) (bool, CloudBaseTransactionCode) {
```

Comment:
Yes thats why I have the code in consensus.go that validates a received mined block and does that check with the address and the block reward to make sure they are equal to the current block reward. The miner should technically be able to set the mining values to whatever they want, as long as the output including address and and reward is valid (1 output and a non 0 address or reward), but the rest of the network needs to confirm that the final mined block adheres to our consensus rules.